components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCommunicationBusContextImpl.java

```
@@ -98,97 +110,88 @@ public RDBMSCommunicationBusContextImpl(DataSource dataSource) {
-     */
-    private void createClusterNodeStatusTable() {
-        Connection connection = null;
-        PreparedStatement preparedStatement = null;
-        try {
-            connection = getConnection();
-            preparedStatement = connection.prepareStatement(RDBMSConstants.CREATE_CLUSTER_NODE_STATUS_TABLE);
-            preparedStatement.execute();
-            connection.commit();
-            if (log.isDebugEnabled()) {
-                log.debug("Cluster Node Status Table Created Successfully");
-            }
-        } catch (SQLException e) {
-            throw new ClusterCoordinationException("Error in executing create cluster node status table query.", e);
-        } finally {
-            close(preparedStatement, "Execute query");
-            close(connection, "Execute query");
-        }
+    private enum Table {
+        LEADER_STATUS_TABLE, CLUSTER_NODE_STATUS_TABLE, MEMBERSHIP_EVENT_TABLE, REMOVED_MEMBERS_TABLE
     }
 
-    /**
-     * Create Membership Event Table.
-     */
-    private void createMembershipEventTable() {
-        Connection connection = null;
-        PreparedStatement preparedStatement = null;
+    private void createTableIfNotExist(Table tableName) {
+        Statement stmt = null;
+        Connection con = null;
         try {
-            connection = getConnection();
-            preparedStatement = connection.prepareStatement(RDBMSConstants.CREATE_MEMBERSHIP_EVENT_TABLE);
-            preparedStatement.execute();
-            connection.commit();
-            if (log.isDebugEnabled()) {
-                log.debug("Membership Event Table Created Successfully");
+            try {
+                con = getConnection();
+                con.setAutoCommit(false);
+                stmt = con.createStatement();
+            } catch (SQLException e) {
+                log.error("Cannot establish connection to the database to create " + tableName + " if not exists", e);
+                return;
```

Why don't we close the connection? Why are we swallowing the exception?

---

This is wrapped in a try finally block, where the connection will be closed.

---

Fixed to throw exception in commit cf4cdfa